### ISSUE TITLE
Transport Quest Tracking Special Behaviour (2nd,8th,12th,18th,22nd,28th day)

### ISSUE THREAD

**Issue by ReiFan49** [KC3Kai/KC3Kai#422]
ok, as per current update that the hunting quest is on the way,

the transport quest have its special behaviour on day with remainder of 2 and 8 from 10 (let's say today for example)
we know that there's an extra quest that appears, which is hunt 5 transport ships.

nah, what i mean is, if i activate one of them, both of them will count separately (so basically you could hunt around 5-10 transport ship by activating the one of the quest itself)
i still can't find out the way the quest tracks out the transport counter, which is (by guessing) looks like swapping each tracker every quest.

(if there's any mechanism change on transport hunting for such certain day, then the tracker mechanism needs to be updated too)

**Comment by bamboo3250** [comment]
The hunting quest is tracked in `Node.js`

![screen shot 2015-07-28 at 4 25 57 pm](https://cloud.githubusercontent.com/assets/8078323/8926834/7460c176-3545-11e5-90a4-d1fefd43691f.png)

Besides the days it is available,  it has no difference from other daily quest. It is still reset at 5AM JST and counted separately. So here it's supposed to work correctly up until this version.

**Comment by ReiFan49** [comment]
wait wait,

Given State: (if activate one of them, as what i said before)
![image](https://cloud.githubusercontent.com/assets/9847780/8927259/5aab426a-3540-11e5-8f18-8a5ca4f57bfc.png)
![image](https://cloud.githubusercontent.com/assets/9847780/8927267/6565f07e-3540-11e5-9ea5-56c6b3340a61.png)

Given Counter:
![image](https://cloud.githubusercontent.com/assets/9847780/8928573/82cc9420-3549-11e5-8cb1-c9fbd0e66ce8.png)

having one of them activated, automatically activates the other one

**Comment by bamboo3250** [comment]
oh my... you mean the game uses the same counter for Bd5 (kill 3 transport ships) and Bd6(kill 5 transport ships)?

**Comment by ReiFan49** [comment]
yes. it is.

**Comment by bamboo3250** [comment]
That's really unexpected. If it is true then our current tracking system cannot apply for this kind of quests. I haven't done any sortie daily quest so I'll investigate it more when I come home. :open_mouth:

**Comment by ReiFan49** [comment]
but, to make sure both of them share the same one or only one way sharing,
the second case need to be run,
which is the opposing state as my picture post.. `activate Bd5, deactivate Bd6` (as today tested with Bd6 towards Bd5, i cannot do the opposing one)

**Comment by bamboo3250** [comment]
okay then I will test the second one, deactivate Bd5 and activate Bd6

**Comment by bamboo3250** [comment]
![screen shot 2015-07-28 at 9 29 36 pm](https://cloud.githubusercontent.com/assets/8078323/8932458/e30a1198-356f-11e5-87b4-a5ec851fc54c.png)

Now thing becomes really weird. 80% of Bd5???? :scream: 2/3 would be 50% and 3/3 should be 100%. 
From what i see, they surely share the same counter but somehow when Bd5 is fulfilled while being deactivated, it shows 80% instead of 100%.

**Comment by ReiFan49** [comment]
(you tested the same case...... `activate Bd6 deactivate Bd5`)

however, if i remember on 3 counter case, 1 for 50%, 2 for 80%, 3 for 100%

**Comment by bamboo3250** [comment]
Actually I tested both cases. Firstly, I activated Bd5 only and killed 2 Transport ships (it showed 50% for Bd5 here). Then switched back to Bd6 and deactivated Bd5 and I killed 2 Transport ships again. I got the above screen (Bd5 became 80%).

**Comment by ReiFan49** [comment]
i'm afraid that Bd5 updates abnormally while Bd6 updates normally (regardless of activation state, only one of them is activated)
like Bd5 counts half of the Bd6 .... or have expanded bound from 3 to 6...

**Comment by bamboo3250** [comment]
Then I think that must be a bug that the KanColle Dev doesn't bother to fix at all because it doesn't make sense to the quest activation system, where the quest should be only counted when activated.

**Comment by dragonjet** [comment]
hm what the status of this? if i understand correctly you discovered a kancolle bug? :open_mouth:

**Comment by bamboo3250** [comment]
Haha for real? :smile: 

Anyway, since this is a bug from Kancolle, I prefer not to follow it. We should keep our tracking system. So you can add tag discuss for this.

**Comment by dragonjet** [comment]
yep, back then even before she was on LSC, there were rumors of Musashi being awarded but I'm not sure if it was a hoax.. The prize for bugs is real though iirc, only the musashi one is not proven.

**Comment by bamboo3250** [comment]
@ReiFan49 Do you want to try first? :sparkles:

**Comment by Crazyteitoku** [comment]
I didn't give much thought on this issue because I sort of knew there was a bug with the game but never took much notice whenever it happened. 

Personally it might even be hard to implement as well considering that not too long ago I encountered a similar case (here)[https://github.com/dragonjet/KC3Kai/issues/422#issuecomment-125611300]. The only exception is my Weekly Quest 1 (Bw1) was already at 80% with the only need to score 4 more A ranks (at least) at boss nodes. Anyway that Bw1 case is besides the point. Maybe if both quests are counted (Bd5 and Bd6) are counted as 80% even though Bd5 already is counted as 3/3 is to perhaps either set it as 2/3 somehow? or maybe not count it entirely and just show percentages done instead? :laughing:

**Comment by ethrundr** [comment]
As far as I remember, there are some bugs regarding to the quests, especially ones with counter as this one. However, since it's minor and not that disturbing, I rather disregard it.

About Bd5 and Bd6, if Bd6 already cleared BEFORE Bd5 is turned on, it'll stuck in 80% with 3/3 sunk count. You'll just need to sunk another one, and the quest will be cleared. I think this happen since Bd5 and Bd6 is sharing the sunk count. However, since Bd5 not yet activated at the moment you cleared Bd6 with 5 sunk count, the Bd5 cannot be marked as clear (just what do you want to mark in the first place if the quest is not active yet). Thus, you get the 80% mark, to indicate that by count you actually already cleared it, but since not yet activated, it cannot be marked as clear. As for the additional sunk, it is to notify the game that you actually already have enough sunk count to get the quest clear. At least this is what I can think so far.

**Comment by dragonjet** [comment]
Related #552 #700

**Comment by Tsubakura** [comment]
Hey guys, I finally got some time on my hand and decided to start investigating in this matter after receiving conflicts about whether this has been fixed or not.
I gathered some people from the Discord and managed to get these samples:

Tsubakura:
4 transport kills on 2-2
Before 0/3 and 0/5
After 3/3 and 4/5

Minako:
3 transport kills on 2-2
Before 0/3 and 0/5
After 3/3 and 4/5

Juu:
2 transport kills on 2-3
Before 0/3 and 0/5
After 3/3 and 4/5

Dusk:
2 transport kills on 2-3
Before 0/3 and 1/5
After 2/3 and 3/5

Chaz:
1 transport kill on 4-3
Before 0/3 and 0/5
After 2/3 and 1/5

Dechi:
1 transport kill on 4-3
Before 0/3 and 1/5
After 3/3 (80%) and 3/5

Basically, the following can be observed:

- The double count bug is not fixed
- The double count bug only works if you have Bd5 and Bd6 active at the same time
- Destroying 2-3-4 transports will always result in 4/5 if you started from 0/5.
- Destroying 2 transports while your progress is 2/5 will also result in 4/5.
- There's no cap stopping Bd5 from going to 3/3 instantly
- The progress counter for Bd5 and Bd6 is shared, even if they are not accessible yet. For example if 1/5 is reached with Bd6 and you make Bd5 active at that point, the game is tricked into thinking that your progress is 1/3 as well, while in reality, its still 0/3. This also explains the weird situations with 80% progress on Bd5. The reverse is most likely also true.
- The trigger for the info feedback is above 50% from what I've heard. This also explains why the sample from Chaz displays 2/3 and 1/5. In reality, it is 2/3 and 2/5 because of the double count, but since its still below 50%, the KC3 didn't pick that up and displays 1/5 as usual.

I am no programmer myself, but I am in the middle of a course atm.
At most, I can only provide you samples and the logical conclusions from it.

Good luck ;)

**Comment by Javran** [comment]
no real new insight about this issue, but in my opinion we don't have to deal with this complication while we already have a mechanism for correcting counters; and in addition to count sunk transport ships, we also have shared counter about sinking subs, and counting PvP wins - I don't feel like we are going to solve them all. and additionally with the release of andoird version, we'll lose track anyway if players make some progress on their phones.
